discovery_engine_core/bindings/src/lib.rs

```diff
@@ -505,7 +505,7 @@ impl XaynDiscoveryEngineAsyncFfi {
 
     /// Disposes the engine.
     pub async fn dispose(engine: Box<SharedEngine>) {
-        engine.as_ref().as_ref().lock().await;
+        drop(engine);
```

Comment:
Any idea why we acquired and directly dropped the lock before dropping the lock with the engine?